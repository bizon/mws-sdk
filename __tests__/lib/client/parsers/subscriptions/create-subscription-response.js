const parseXml = require('../../../../../lib/client/parsers')
const parseCreateSubscriptionResponse = require('../../../../../lib/client/parsers/subscriptions/create-subscription-response')

describe('lib.client.parsers.subscriptions.create-subscription-response', () => {
  it('should parse the CreateSubscriptionResponse example response from MWS doc', () => {
    const doc = parseXml(
      `<?xml version="1.0"?>
      <CreateSubscriptionResponse xmlns="http://mws.amazonservices.com/schema/Subscriptions/2013-07-01">
        <CreateSubscriptionResult />
        <ResponseMetadata>
          <RequestId>c9bb2e77-2425-4e1a-9c85-36d00EXAMPLE</RequestId>
        </ResponseMetadata>
      </CreateSubscriptionResponse>`,
    )

    const res = parseCreateSubscriptionResponse('/subscriptions:CreateSubscriptionResponse', doc)

    expect(res).toMatchSnapshot()
  })
})
