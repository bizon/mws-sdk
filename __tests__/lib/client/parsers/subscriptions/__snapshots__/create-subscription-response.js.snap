// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib.client.parsers.subscriptions.create-subscription-response should parse the CreateSubscriptionResponse example response from MWS doc 1`] = `
{
  "createSubscriptionResult": {},
  "responseMetadata": {
    "requestId": "c9bb2e77-2425-4e1a-9c85-36d00EXAMPLE",
  },
}
`;
